src/net/dmytro/homework4/Animal.java

```diff
@@ -1,10 +1,24 @@
 package net.dmytro.homework4;
 
 public abstract class Animal {
-  protected int maxSwim;
-  protected int maxRan;
+  protected int maxSwimDistance;
+  protected int maxRanDistance;
+  protected int distanceToDo;
+  private static int count;
 
+  public Animal(){
+    count++;
+  }
+  public static int getCount(){
+    return count;
+  }
+
+
+
+
+
+
+    abstract void ran(int maxRanDistance, int distanceToDo);
```

Comment:
maxDistance shouldn't be parameter, it should be class field, and every animal has different value.  

for example 
`new Dog("Tuzik", 500, 20); `

 or for cat, as we know cats cannot swim. 
 `new Cat("Murzik", 30);`